CHANGELOG.md

```diff
@@ -3,6 +3,18 @@
 ## Unreleased changes
 
 - Extend enum `ProtocolVersion` enum with a protocol version 7 variant `PROTOCOL_VERSION_7`.
+- Support for changes to cooldown behavior in protocol version 7:
+  - `AccountInfo` has a new repeated `cooldowns` field. Each `Cooldown` records
+    the amount, (expected) release time and whether it's a regular cooldown,
+    pre-cooldown or pre-pre-cooldown.
+  - `AccountInfo` now includes `available_balance`. This is included since the
+    method for calculating it has changed (it now must account for the
+    cooldowns), so this is provided as a convenience.
```

Comment:
I think it's problematic to include redundant data like this, especially if it is used for calculations and checks. Wouldn't you have to verify that the available balance is correct by calculating it anyway? I mean can you *trust* that the field is always correct and not maliciously set to a wrong value?

Even if you can trust it, I still don't like redundant data.

Reply:
I think it's reasonable to avoid redundant data generally, but this is a case where I think there's a benefit:

- It's become increasingly complicated to compute the available balance. Having it available in the API is likely to be useful.
- If you're concerned about the source of the data (the node) being untrustworthy, then there's no reason to rely on any of the other data about the account that you would use to calculate the available balance.
- It is possible that the available balance calculation will change again in future (for instance, with the introduction of jailing). Having the available balance present in the API makes it easier for applications to be somewhat futureproof in the event of such changes.

Reply:
That sounds fair, thanks for the justification :+1: